src/doc/trpl/const.md

```diff
@@ -1,3 +1,25 @@
 % `const`
 
-Coming soon!
+Rust has a way of defining constants with the `const` keyword:
+
+```rust
+const N: i32 = 5;
+```
+
+Unlike [`let`][let] bindings, you must annotate the type of a `const`.
+
+[let]: variable-bindings.html
+
+Constants live for the entire lifetime of a program, and therefore any
+reference stored in a constant has a [`'static` lifetime][lifetimes]:
+
+```rust
+const NAME: &'static str = "Steve";
+```
+
+[lifetimes]: lifetimes.html
+
+More specifically, constants in Rust have no fixed address in memory. This is
+because they’re effectively inlined to each place that they’re used. References
```

Comment:
This sentence kinda contradicts the above claim that they can store any reference with a `'static` lifetime I think. I think that this may want to remove the reference to the `'static` lifetime and focus on the constant/inlining aspect.

Constants should be used for values that do not need an address in memory (as you've said), but they are allowed to reference addresses (e.g. the `&'static str` case), but that's mostly an implementation detail that the contents of the reference are stored in static memory somewhere.